cmake/find/FindGMP.cmake

```diff
@@ -0,0 +1,76 @@
+# Try to find the GNU Multiple Precision Arithmetic Library (GMP)
```

Comment:
GMP/MPFR are a pain yes. We are not building them from source since they do not use a CMake build system. So the only option is to install via a package manager on macOS/Linux, or download a pre-built dll on Windows (which this find_package() will find).